### DISCUSSION TITLE
Trunk and DDoS  attacks

### DISCUSSION THREAD

**Discussion by SamtheTruck** [yggdrasil-network/yggdrasil-go#1027] [Q&A] [answered]
First, I think the ideas you have are brilliant, really useful and necessary. Extremely useful. I particularly like the idea that a very small number of addresses or DHT can route traffic all over. Very crafty. It reminds me of how small, limited systems of rules can lead to very complex systems. Like DNA or the game of life, etc. It really caught my eye.

I have often wondered about the above, (Trunk and DDoS  attacks), bringing the whole network to a halt. Maybe you've described this elsewhere but if you have it’s not in a way I can understand nor recognize. So...

1. What happens if someone takes control of a very large amount of the trunk nodes and screws them up?

2. I notice here,

https://yggdrasil-network.github.io/2021/06/19/preparing-for-v0-4.html

it says,

"...The spanning tree works largely the same way as before. The only significant differences are with the root selection and updates: the root is the node with the lowest ed25519 public key, rather than the highest sha512sum hash of the public key, and the root updates the timestamp for its spanning tree announcements every 30 minutes ..."

How can this not be spoofed so that the root is controlled by hostiles?
And does this mean the trunk addresses mutate and change around every 30 minutes as stated in the link?

3. Is there, or is there, any plan, to use a network wide anti-DDoS attack recognition subroutine? Could they be shut down or ignored if they are spewing out a ton of request? It seems they could be ignored after a certain threshold of repeating request is reached. There being a "trunk" it would seem to be obvious to the "trunk" of a vast amount of repeating request coming through.

4 Another thing I don't understand, and you have probably explained, is how since the trunk is a major point of entry there's not a choke point there that limits traffic.

Pardon me if my questions are stupid and obvious. Sometimes you have ask, even at the risk of being the public buffoon.

**Comment by majestrate** [comment, marked as answer]
if the trunk is a bad node the network takes a shit. this has happened before except "accidentally". someone was generating very low yggdrasil addresses and one of the smaller unreliable ones ended up becoming the root node.

**Reply by SamtheTruck** [reply]
Thank you for the answer. I was afraid of that. Sigh. Then there appears no way to keep (them), whoever (them) may be, from blocking.

**Reply by Cuteistfox** [reply]
why not mark it as ansewrd
